### ISSUE TITLE
Pandas progress bar when iterating over rows

### ISSUE THREAD

**Issue by Aorus42** [tqdm/tqdm#322]
Maybe this is already taken care of and I just don't know how to use the pandas implementation correctly, but the tqdm progress bar returns an incorrect total when running apply(axis=1), i.e. applying a function to each row of a dataframe. I have implemented the following fix for myself, but could something similar be added to the official code? Thanks!

```
# Precompute total iterations
if kwargs.get('axis', 0) == 1:
    total = len(df)
else:
    total = getattr(df, 'ngroups', None)
    if total is None:  # not grouped
        total = len(df) if isinstance(df, Series) \
            else df.size // len(df)
    else:
        total += 1  # pandas calls update once too many
```

**Comment by casperdcl** [comment]
hopefully fixed by #299?

**Comment by lrq3000** [comment]
Yes Casper is right, this is exactly the sort of issue the mentioned PR
should fix. Please try it and report if the bug is fixed in your case!

Le 10 Déc. 2016 12:10, "Casper da Costa-Luis" <notifications@github.com> a
écrit :

**Comment by lrq3000** [comment]
I think we should merge #299 because it should fix this kind of issues, not perfectly, but at least it will make tqdm_pandas work for both rows and colums calculations. Also it *might* also fix calculations for higher dimension data structures like Panels.

**Comment by soedr** [comment]
Any update on this? Just re-installed `tqdm` with `pip install tqdm --upgrade`  and am getting an empty progress bar iterating over columns. 
   
    from tqdm import tqdm, tqdm_pandas
    import pandas as pd

    data = pd.DataFrame({'foo':[1,2,3,4,5,6,7,8,9,10],'bar':[1,2,3,1,2,3,1,2,3,1]})

    tqdm.pandas(tqdm())

    def my_func(df):
        new_number = df['foo'] * df['bar']
        return new_number

    data['new_number'] = data.progress_apply(my_func, axis = 1)

Which prints:

      0%|          | 0/2 [00:00<?, ?it/s]
      10it [00:00, 4456.34it/s]

**Comment by casperdcl** [comment]
Hi @soedr, we were waiting for someone to fix #299. Personally I don't use pandas, and don't really have time to fix it.

**Comment by casperdcl** [comment]
fixed in #524
